keps/sig-network/1880-multiple-service-cidrs/README.md

```diff
@@ -0,0 +1,975 @@
+    log
+    delete all subsets, leaving the largest set that exactly matches on at least on family
+  endif
+ endif
+ if kubernetes.default does not exist
+  create it
+```
+
+#### The special "default" ServiceCIDRConfig
+
+The `kubernetes.default` Service is expected to be covered by a valid range. Each apiserver will
+ensure that a ServiceCIDRConfig object exists to cover its own flag-defined ranges, so this should
+be true in normal cases. If someone were to force-delete the ServiceCIDRConfig covering `kubernetes.default` it
+would be treated the same as any Service in the repair loop, which will generate warnings about
+orphaned Service IPs.
+
+#### Service IP Allocation
+
+When a a new Service is going to be created and already has defined the ClusterIPs, the allocator will
+just try to create the corresponding IPAddress objects, any error creating the IPAddress object will
+cause an error on the Service creation. The allocator will also set the reference to the Service on the
+IPAddress object.
+
+The new allocator will have a local list with all the Service CIDRs and IPs already allocated, so it will
+have to check just for one free IP in any of these ranges and try to create the object.
+There can be races with 2 allocators trying to allocate the same IP, but the storage guarantees the consistency
+and the first will win, meanwhile the other will have to retry.
+
+Another racy situation is when the allocator is full and an IPAddress is deleted but the allocator takes
+some time to receive the notification. One solution can be to perform a List to obtain current state, but
+it will be simpler to just fail to create the Service and ask the user to try to create the Service again.
+
+If the apiserver crashes during a Service create operation, the IPAddress is allocated but the
+Service is not created, the IPaddress will be orphan. To avoid leaks, a controller will use the
+`metadata.creationTimestamp` field to detect orphan objects and delete them.
+
+There is going to be a controller to avoid leaking resources:
+
+- checking that the corresponding parentReference on the IPAddress match the corresponding Service
+- cleaning all IPAddress without an owner reference or, if the time since it was created is greater
+than 60 seconds (default timeout value on the kube-apiserver )
+
+#### Service IP Reservation
+
+In [keps/sig-network/3070-reserved-service-ip-range](https://github.com/kubernetes/kubernetes/issues/95570) a new feature was introduced that allow to prioritize one part
+of the Service CIDR for dynamically allocation, the range size for dynamic allocation depends
+ont the size of the CIDR.
+
+The new allocation logic has to be compatible, but in this case we have more flexibility and
+there are different possibilities, that should be resolved before Beta.
+
+```
+<<[UNRESOLVED keps/sig-network/3070-reserved-service-ip-range]>>
+Option 1: Maintain the same formula and behavior per ServiceCIDRConfig
```

Comment:
FWIW I think we should do option 1 (no API) and see if we really need option 3.  I suspect the impl will be very similar.

Reply:
"less is more" thing, right?